src/pages/Appearance/appearanceDatastoreQuery.js

```diff
@@ -0,0 +1,51 @@
+import { NAMESPACE } from '../../constants/data-store'
+import { useDataQuery } from '@dhis2/app-runtime'
+
+const APPEARANCE = 'appearance'
+
+/**
+ * update data store
+ * key: Appearance
+ * Settings that can be saved here: Home Screen, Programs, Dataset
+ * */
+
+export const saveAppearanceKeyMutation = {
+    resource: `dataStore/${NAMESPACE}/${APPEARANCE}`,
+    type: 'update',
+    data: ({ settings }) => ({
+        ...settings,
+        lastUpdated: new Date().toJSON(),
+    }),
+}
+
+/**
+ * Query to get Appearance Settings from Data store
+ * key: Appearance Settings
+ * */
+
+export const getAppearanceKeyQuery = {
+    appearanceSettings: {
+        resource: `dataStore/${NAMESPACE}/${APPEARANCE}`,
+    },
+}
+
+/**
+ * Read Appearance Data store values:
+ *
+ * */
+
+export const useReadAppearanceDataStore = () => {
+    const { loading, error, data } = useDataQuery(getAppearanceKeyQuery)
+
+    return {
+        load: loading,
+        error,
+        completionSpinner: data && data.appearanceSettings.completionSpinner,
```

Comment:
You should be able to use this notation:

`data?.appearanceSettings?.completionSpinner`

You want the values to be undefined if the property doesn't exist? (just confirming)